client.go

```diff
@@ -322,6 +325,14 @@ func WithAPIEndpoint(endpoint string) ClientOptions {
 	}
 }
 
+// WithTerraformProvider allows to pass in the Terraform version utilized by
```

Comment:
nit: maybe we could update this comment to more directly describe what we are doing here?  Something like: "WithTerraformProvider configures the client to be used as the PagerDuty Terraform provider"